_field-types/supported-field-types/wildcard.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: default
+title: Wildcard
+nav_order: 62
+has_children: false
+parent: String field types
+grand_parent: Supported field types
+---
+
+# Wildcard field type
+
+A `wildcard` field is a variant of a `keyword` field designed for arbitrary substring and regular expression matching.
+
+Use a `wildcard` field when your content consists of "strings of characters" and not "text". Examples include unstructured log lines and computer code.
+
+The `wildcard` field type is indexed differently from the `keyword` field type. Whereas `keyword` fields write the original field value to the index, the `wildcard` field type splits the field value into substrings with a length that is less than or equal to 3 and writes the substrings to the index. For example, the string `test` is split into strings `t`, `te`, `tes`, `e`, `es`, and `est`. At search time, required substrings from the query pattern are matched against the index to produce candidate documents, which are then filtered according to the pattern in the query.
```

Comment:
Specifically, at search time, if we query for the term `test` we do an indexed search for `tes AND est`. (We only use the 1-2 character substrings when the search term is shorter than 3 characters.)

For each matching document, we check if the source value is `test` (to exclude false positive values like `nikola tesla felt alternating current was best`).